src/ocad-reader/object-index.js

```diff
@@ -4,7 +4,12 @@ const TObject = require('./tobject')
 module.exports = class ObjectIndex {
   constructor(reader, version) {
     this.version = version
-    this.nextObjectIndexBlock = reader.readInteger()
+
+    // Ignore pointers that do not point to a valid location in the file.
+    // Compare getBlockCheckedRaw() in Open Orienteering Mapper.
+    this.nextObjectIndexBlock = (i =>
+      i > reader.buffer.length - (256 * 40 + 4) ? 0 : i)(reader.readInteger())
+
```

Comment:
As mentioned in #20, I took this change and made a slight variation to also add a warning when we do this.

Reply:
This was the first obstacle I encountered when trying to read an Ocad course setting file. I had to setup the devenv and debug Openorienteering mapper to understand that the next block pointer was just gibberish.